src/otaclient_common/linux.py

```diff
@@ -252,3 +258,33 @@ def copyfile_nocache(src: StrOrPath, dst: StrOrPath) -> None:
         finally:
             os.posix_fadvise(src_fd, 0, 0, os.POSIX_FADV_DONTNEED)
             os.posix_fadvise(dst_fd, 0, 0, os.POSIX_FADV_DONTNEED)
+
+
+#
+# ------ fstrim with timeout ------ #
+#
+
+
+def fstrim_at_thread(
+    target_mountpoint: Path, *, waited: bool = True, timeout: float = 120
+) -> bool:  # pragma: no cover
+    """Dispatch a sub-thread for running fstrim with timeout.
+
+    fstrim is safe to be interrupted at any time.
+
+    Returns:
+        True if fstrim finished without reaching timeout.
+    """
+
+    def _thread_entry():
+        _cmd = ["fstrim", "-v", str(target_mountpoint)]
+        subprocess.run(_cmd, timeout=timeout)
+
+    _t = threading.Thread(target=_thread_entry, name="otaclient_fstrim", daemon=True)
+    try:
+        _t.run()
```

Comment:
According to the comment, the intention is to execute as another thread?
```suggestion
        _t.start()
```

Reply:
!! Oh no, that is a mistake!
`run` actually directly running the target method in current thread instead of spawning a new thread :face_with_head_bandage: , let me fix that ...